src/io/iter_image_recordio_2.cc

```diff
@@ -133,15 +134,14 @@ inline void ImageRecordIOParser2<DType>::Init(
   record_param_.InitAllowUnknown(kwargs);
   batch_param_.InitAllowUnknown(kwargs);
   normalize_param_.InitAllowUnknown(kwargs);
-  prefetch_param_.InitAllowUnknown(kwargs);
   n_parsed_ = 0;
   overflow = false;
   rnd_.seed(kRandMagic + record_param_.seed);
   int maxthread, threadget;
   #pragma omp parallel
   {
     // be conservative, set number of real cores
-    maxthread = std::max(omp_get_num_procs() / 2 - 1, 1);
+    maxthread = std::max(omp_get_num_procs(), 1);
```

Comment:
Does this get the # of logic cores?

Reply:
Yes. I think we should let user to make the decision, just as normal operator.  Before this change, data loader will only use n/2 cores for `export OMP_NUM_THREADS=n`.

Reply:
Make sense

Reply:
Since now the iterator is pushed to engine, will this omp threading make troubles?